src/bespokelabs/curator/request_processor/base_request_processor.py

```diff
@@ -223,10 +229,17 @@ def create_dataset_files(
         responses_files = glob.glob(f"{working_dir}/responses_*.jsonl")
         if len(responses_files) == 0:
             raise ValueError(f"No responses files found in {working_dir}")
-        dataset_file = f"{working_dir}/dataset.arrow"
+        dataset_file = f"{working_dir}/{parse_func_hash}.arrow"
         if os.path.exists(dataset_file):
             logger.info(f"Using existing dataset file {dataset_file}")
-            return Dataset.from_file(dataset_file)
+            try:
+                output_dataset = Dataset.from_file(dataset_file)
+                return output_dataset
+            except Exception as e:
```

Comment:
We shouldn't catch broad exceptions. It can mask many issues.
For example if there was a type (return otput_dataset), that will throw an exception which will be caught here.

Reply:
Thinking about this more, we probably shouldn't remove the file here, but raise a ValueError similar to what I suggested in the comment below

 f"Failed to load dataset from {dataset_file}. This was likely corrupted by a failed previous run. You can delete this file and try again to see if the error is transient."

Reply:
@madiator The user shouldn't have to know about the dataset file since it is in the cache. If you are using Prompter and it starts complaining about a dataset file, you would probably be confused and wouldn't know what to do right? At least adding an additional comment about the relation to parse_func helps put people on the track of fixing it.

Reply:
I moved the return out of the try catch, but left the naked exception handling. 
I think since it is at the top of the logic, just deleting and retrying makes sense to me. If it fails again, the next `Dataset.from_file()` call only catches `pyarrow.lib.ArrowInvalid` so the  Exception will be surfaced again if there still is an issue. 

Also OK to change this naked exception to `pyarrow.lib.ArrowInvalid`, but just trying to help the user not touch the cache.